src/components/shared/LanguagesList.vue

```diff
@@ -46,36 +41,65 @@ export default {
     return {
       val: 0,
       selectedLanguage: '',
+      keepFirst: true,
+      openOnFocus: true,
+      selected: null,
     };
   },
   computed: {
     languages() {
       if (this.customLanguages) {
-        return [{ id: 0, title: `${this.title}`, active: this.current === 0 }]
-          .concat(this.customLanguages);
+        return [].concat(this.customLanguages);
       }
-
-      return [{ id: 0, title: `${this.title}`, active: this.current === 0 }]
+      return []
         .concat(Object.keys(LANGUAGES)
           .map((lang, index) => ({ id: index + 1, title: LANGUAGES[lang], value: lang })));
     },
-    dropdownTitle() {
-      return this.languages.reduce((current, language) => (
-        this.val > 0 && language.id === this.val
-          ? language.title
-          : current), this.languages[0].title);
+    filterLanguages() {
+      return this.languages.filter(option => option.title
+        .toString()
+        .toLowerCase()
+        .indexOf(this.selectedLanguage.toLowerCase()) >= 0);
+    },
+    optionsLanguages() {
+      return this.filterLanguages.map(option => option.title);
     },
   },
   watch: {
-    selectedLanguage(value) {
+    selected(value) {
       this.$emit('input', value);
     },
+    selectedLanguage() {
+      if (this.selectedLanguage === '') {
+        this.val = 0;
+        this.selected = null;
+      }
+    },
   },
   methods: {
-    select(id, language) {
-      this.val = id;
-      this.selectedLanguage = language;
+    select(option) {
+      this.languages.filter((language) => {
+        if (language.title === option) {
+          this.selected = language.value;
+          this.val = language.id;
+        }
+        return '';
+      });
```

Comment:
O que faz esse filter?

Reply:
Busca o texto que é digitado no input, dentro do array de palavras. Faz o efeito de quando for digitada alguma palavra ir aparecendo as opções parecida

Reply:
Esse é o outro filter, não? Esse parece que serve pra achar o idioma selecionado no array. Não seria melhor usar um `find` pra não precisar percorrer o array todo?

Reply:
O find não deveria ser usado para encontrar apenas um único elemento ? se o usúario digitar o começo de uma palavra que tenha outras parecida ele ira retorna apenas a primeiro que entrar, não?

Reply:
Esse filter que tô falando é quando seleciona uma opção no autocomplete, pra mudar o selected